#Lab Notebook for Ecological Genomics

### Authors: Samantha Alger
###University of Vermont
## Date started: (2017-01-21)
## Date end:


# Table of contents (for 200 entries)    
* [Page 1: 2017-01-23](#id-section1). Next Gen Sequencing
* [Page 2: 2017-01-25](#id-section2). QTNs and Project ideas
* [Page 3: 2017-01-30](#id-section3). Project ideas
* [Page 4: 2017-02-01](#id-section4). Genome sequencing types
* [Page 5: 2017-02-01](#id-section5). Getting onto Unix Server, Terminal Commands
* [Page 6: 2017-02-07](#id-section6). RNA-seq Tutorial
* [Page 7: 2017-02-08](#id-section7). Transcriptomics
* [Page 8 2017-02-13:](#id-section8).Transcriptomics
* [Page 9 2017-02-15:](#id-section9). Extract expression
* [Page 10:](#id-section10).
* [Page 11:](#id-section11).
* [Page 12:](#id-section12).
* [Page 13:](#id-section13).
* [Page 14:](#id-section14).
* [Page 15:](#id-section15).
* [Page 16:](#id-section16).
* [Page 17:](#id-section17).
* [Page 18:](#id-section18).
* [Page 19:](#id-section19).
* [Page 20:](#id-section20).
* [Page 21:](#id-section21).
* [Page 22:](#id-section22).
* [Page 23:](#id-section23).
* [Page 24:](#id-section24).
* [Page 25:](#id-section25).
* [Page 26:](#id-section26).
* [Page 27:](#id-section27).
* [Page 28:](#id-section28).
* [Page 29:](#id-section29).
* [Page 30:](#id-section30).
* [Page 31:](#id-section31).
* [Page 32:](#id-section32).
* [Page 33:](#id-section33).
* [Page 34:](#id-section34).
* [Page 35:](#id-section35).
* [Page 36:](#id-section36).
* [Page 37:](#id-section37).
* [Page 38:](#id-section38).
* [Page 39:](#id-section39).
* [Page 40:](#id-section40).
* [Page 41:](#id-section41).
* [Page 42:](#id-section42).
* [Page 43:](#id-section43).
* [Page 44:](#id-section44).
* [Page 45:](#id-section45).
* [Page 46:](#id-section46).
* [Page 47:](#id-section47).
* [Page 48:](#id-section48).
* [Page 49:](#id-section49).
* [Page 50:](#id-section50).
* [Page 51:](#id-section51).
* [Page 52:](#id-section52).
* [Page 53:](#id-section53).
* [Page 54:](#id-section54).
* [Page 55:](#id-section55).
* [Page 56:](#id-section56).
* [Page 57:](#id-section57).
* [Page 58:](#id-section58).
* [Page 59:](#id-section59).
* [Page 60:](#id-section60).
* [Page 61:](#id-section61).
* [Page 62:](#id-section62).
* [Page 63:](#id-section63).
* [Page 64:](#id-section64).
* [Page 65:](#id-section65).
* [Page 66:](#id-section66).
* [Page 67:](#id-section67).
* [Page 68:](#id-section68).
* [Page 69:](#id-section69).
* [Page 70:](#id-section70).
* [Page 71:](#id-section71).
* [Page 72:](#id-section72).
* [Page 73:](#id-section73).
* [Page 74:](#id-section74).
* [Page 75:](#id-section75).
* [Page 76:](#id-section76).
* [Page 77:](#id-section77).
* [Page 78:](#id-section78).
* [Page 79:](#id-section79).
* [Page 80:](#id-section80).
* [Page 81:](#id-section81).
* [Page 82:](#id-section82).
* [Page 83:](#id-section83).
* [Page 84:](#id-section84).
* [Page 85:](#id-section85).
* [Page 86:](#id-section86).
* [Page 87:](#id-section87).
* [Page 88:](#id-section88).
* [Page 89:](#id-section89).
* [Page 90:](#id-section90).
* [Page 91:](#id-section91).
* [Page 92:](#id-section92).
* [Page 93:](#id-section93).
* [Page 94:](#id-section94).
* [Page 95:](#id-section95).
* [Page 96:](#id-section96).
* [Page 97:](#id-section97).
* [Page 98:](#id-section98).
* [Page 99:](#id-section99).
* [Page 100:](#id-section100).
* [Page 101:](#id-section101).
* [Page 102:](#id-section102).
* [Page 103:](#id-section103).
* [Page 104:](#id-section104).
* [Page 105:](#id-section105).
* [Page 106:](#id-section106).
* [Page 107:](#id-section107).
* [Page 108:](#id-section108).
* [Page 109:](#id-section109).
* [Page 110:](#id-section110).
* [Page 111:](#id-section111).
* [Page 112:](#id-section112).
* [Page 113:](#id-section113).
* [Page 114:](#id-section114).
* [Page 115:](#id-section115).
* [Page 116:](#id-section116).
* [Page 117:](#id-section117).
* [Page 118:](#id-section118).
* [Page 119:](#id-section119).
* [Page 120:](#id-section120).
* [Page 121:](#id-section121).
* [Page 122:](#id-section122).
* [Page 123:](#id-section123).
* [Page 124:](#id-section124).
* [Page 125:](#id-section125).
* [Page 126:](#id-section126).
* [Page 127:](#id-section127).
* [Page 128:](#id-section128).
* [Page 129:](#id-section129).
* [Page 130:](#id-section130).
* [Page 131:](#id-section131).
* [Page 132:](#id-section132).
* [Page 133:](#id-section133).
* [Page 134:](#id-section134).
* [Page 135:](#id-section135).
* [Page 136:](#id-section136).
* [Page 137:](#id-section137).
* [Page 138:](#id-section138).
* [Page 139:](#id-section139).
* [Page 140:](#id-section140).
* [Page 141:](#id-section141).
* [Page 142:](#id-section142).
* [Page 143:](#id-section143).
* [Page 144:](#id-section144).
* [Page 145:](#id-section145).
* [Page 146:](#id-section146).
* [Page 147:](#id-section147).
* [Page 148:](#id-section148).
* [Page 149:](#id-section149).
* [Page 150:](#id-section150).
* [Page 151:](#id-section151).
* [Page 152:](#id-section152).
* [Page 153:](#id-section153).
* [Page 154:](#id-section154).
* [Page 155:](#id-section155).
* [Page 156:](#id-section156).
* [Page 157:](#id-section157).
* [Page 158:](#id-section158).
* [Page 159:](#id-section159).
* [Page 160:](#id-section160).
* [Page 161:](#id-section161).
* [Page 162:](#id-section162).
* [Page 163:](#id-section163).
* [Page 164:](#id-section164).
* [Page 165:](#id-section165).
* [Page 166:](#id-section166).
* [Page 167:](#id-section167).
* [Page 168:](#id-section168).
* [Page 169:](#id-section169).
* [Page 170:](#id-section170).
* [Page 171:](#id-section171).
* [Page 172:](#id-section172).
* [Page 173:](#id-section173).
* [Page 174:](#id-section174).
* [Page 175:](#id-section175).
* [Page 176:](#id-section176).
* [Page 177:](#id-section177).
* [Page 178:](#id-section178).
* [Page 179:](#id-section179).
* [Page 180:](#id-section180).
* [Page 181:](#id-section181).
* [Page 182:](#id-section182).
* [Page 183:](#id-section183).
* [Page 184:](#id-section184).
* [Page 185:](#id-section185).
* [Page 186:](#id-section186).
* [Page 187:](#id-section187).
* [Page 188:](#id-section188).
* [Page 189:](#id-section189).
* [Page 190:](#id-section190).
* [Page 191:](#id-section191).
* [Page 192:](#id-section192).
* [Page 193:](#id-section193).
* [Page 194:](#id-section194).
* [Page 195:](#id-section195).
* [Page 196:](#id-section196).
* [Page 197:](#id-section197).
* [Page 198:](#id-section198).
* [Page 199:](#id-section199).
* [Page 200:](#id-section200).


------
<div id='id-section1'/>
<<<<<<< HEAD
### Page 1: 2017-01-23

### Next Gen Sequencing

**Glossary:** 

- short (50 bp)
- long (100 bp, 150, 300 bp -Hi-seq)
- 10,000-60,000 bp (SMART)
- single vs. paired end
- "Reduce representation": RNA, GBS/RAD seq, near restriction sites

**OUTLINE**
​    

- Advances in Seq Tech
- Range of Applications (Whole Genome Sequencing (WGS), RNAseq, ChipSeq-chromatin immunization sequencing, targeted/capture seq. (use probes that target something)) Whatever the application, probably using Illumina sequencing- sequencing by synthesis (90% of the global data).
- General Library Prep. Workflow
- Sequencing-by-synthesis (SBS)
- Other Technologies
- Learning Activity

**Human Genome Project**
2001-2003

**ABI-Sanger**       
-15 years     
-1 genome   
-$3 B     

**Hi Seq X Ten releases**      
-2014      
-Illumina      
-1 day  
-45 whole genomes human   
-$1000 each   

**What technique you choose depends on what your question is**  
-where is genetic variation 
-phenotypes 
-#samples 
-population v. individual 
-comparative studies  
-model or not?  
-demographic history  
-adaptive genetic variation 
-gene expression var. 

Major decision points: Length of reads, and number of reads, and distribution

WorkFlow: Extraction to get DNA or RNA, if RNA- change into cDNA, fragment sample, ligate adaptors (individual barcodes), add seq. adaptors, PCR).

Extracted DWN with adaptors and put into a lane. bridge amplification, cluster gen. labeled dNTPs build on and a snap shot is taken everytime one is added. 

"den novo" assembly"- with non-model organism.

------
<div id='id-section2'/>
### Page 2: 2017-01-25 

### QTNs 

### Outline:

1. What are QTNs?
2. Quantitative genetic theory of adaptive traits?

- V~a~
- h^2^

1. Methods

- linkage mapping

- GWAS

  - selection scams

  QTN = "Qunatitative trait nucleotides", most simple SNPS
  Study of **quantiative traits**- traits that have continuous distributions, traits have a mean and variance. 

```
variance between the two allehels to determine the "average effect"
```

```
* flowering time- quantitative trait
* flower color- Mendelian trait -discrete
* thermal tolerance
* venom potency
* defense compounds
* drought tolerance
* altitude tolerance (hypoxia)
```

```
Most populations are at their adaptive peak- because of this, selection on a small mutation might move fitness up or down. selection on a large mutation, would most likely move fitness down- therefore, most of selection occurs on small mutations because large mutations are usually  deleterious. 
```

```
​
```

## Seastar wasting disease overview:

- High morality rate and affects many species. East and West coast.    
- What is causing it?-unknown  
  - In hours or days, a healthy individual with lose legs, innards expload out  
  - 2012 first account, 2014- really bad, still occuring but less severe. There have been reports of seastar wasting in 1970s.  
  - Densovirus implicated (Hewsonet al., 2015, PNAS) (SSDNA virus)  
  - coauthors are not convinced that this is causal pathogen.  
  - also present in 70 year old museum specimens.  
  - It could be that something gets out of balance in their normal microbiome and allows the pathogen to make the animal sick.   
  - Finding that cause of disease is a challenge.  
  - Field sampling and lab studies  

## Melanie's experiment:

- take biopsy, exracted total RNA, polyA tail to selected mRNA (to get just genes that are coding for proteins), sequenced on hi-illumina, amplified sequences, (get results for microbes on skin, 16S data), photo taken before or after biopsy

## Questions for Seastar wasting:

1. What is the role of host microbiota in disease prev?
2. How does expression of immune related genes differ between the sick vs. health?
   Can understanding something about the immune response tell us something about what type of pathogen we are dealing with? for example: RNAi is used in antiviral defense. 

------
<div id='id-section3'/>
### Page 3: 2017-01-30

## Project ideas/overview:

Targetted approach to see differences in immune related genes over time- as 'infection progressed'. 
Also look for differences between sick v. healthy individuals. 

H vs. S

Jonathan Rast- reviews on sea urchin immunity. 
Adaptive/innate immune system: 220 Toll- like reptors
vs. human (20 receptors) 

Methods:
Identify genes related to immune response to particular pathogens. Pair with random group of genes to see if there really is upregulation or if it is an artifact. 
Find groups of sick individuals and look at the dataset associated with when each individuals go from healthy to sick.
Pair with healthy individuals at the same time points. 

HH: 5
SS: 5
HS: 5
x two time steps
= 30 data sets

Can we tell anything about the pathogen identity based on expression of immune genes?
What is the immune response? Is it different between 2 spp. of starfish?

Blast against other infected seastars to be confident that these seastars are infected with Densovirus.

------
<div id='id-section4'/>
### Page 4: 2017-02-01

## Genome sequencing types

homework:

- Send link of GitHub notebook to Andrew
- Project proposal due to Melissa and Stephen next Monday:
  - Hypothesis, and how we plan to test it.
- Sign up for 1 info update and 1 discussion

## Whole Genome Sequencing

- applications:
  - high power resolution
    - population genetics
    - conservation- control breeding programs etc. 
    - screen for variation and adaptive potential 
    - impacts of genetic variation
    - plastic responses to the environment
    - local adaptation
  - sometimes you need reference genome- other times you don't: 
    - you **don't need** it with: De novo assembly, adaptations, gene expression patterns, 
    - you **need** it with: select important variation, epigenetic modification, DNA-protein, gene expression
- Prior considerations:
  - $
  - Computational
    - geater than 1 T byte
    - Server
    - command line 
    - Python/perl
- Limitations:
  - polymorphic genes- core genes that are highly conserved
  - Paralogs
  - rapidly evolving genes
  - large gene families- poor representation
  - 1 individual
  - pooled samples
  - Impossible to sequence all nucleotides (not actually Whole genome sequencing) such as highly repetitive regions and heterchromatic regions such as centermeres/telomeres. DNA might be so bound up, highly repetitive stuff cannot be placed onto the map. 
- Genome is a working hypothesis


- Sequencing Platforms
  - Short reads
    - Ilumina sequencing (150 bp)
    - SOLID (50bp)
  - Long Reads
    - Pacific Biosciences (5kb)
    - Ion Torrent (500 bp)
    - Ilumina Mollecule (up to 10 kb)
- knowledge on organism
  - genome size (K-mer approach: short, unique element of DNA sequence of length K).
  - **HOW MANY** reads you need depends on size of the genome. There are some rules of thumb depending on size of the genome. 


- Methods:

  - Wet-lab

    - Tissue
      - high quality: energetically active (mDNA)
      - avoid Gut and Skin- other DNA from symbionts or bacteria will be sequenced
    - Quantity 1 mg -> 6 ug (short)

  - Library Prep. 

    - reads would occur in any direction (single end)

    - paired end reads

    - mate-pair

    - contigs

    - scaffold

      ​

  - Gc Content

  - repeat abundance

  - Duplicate reads

- 50% of the reads come from contigs

- Annotate your sequence (use from related genome)

- Find them in NCBI

## RNA Seq:

- Advantages-

  - Difference gene expressions

  - allele specific expression- environmental response, adaptation

  - function relevant subset of the genome

    ​

    | RNA sequencing                 | Micro-Array                              |
    | ------------------------------ | ---------------------------------------- |
    | wide range of expression value | X                                        |
    | X                              | saturation of analog type fluorescent signal |
    | info on splicing events        | X                                        |

- Limitations to RNA seq

  - Can tell you about differential gene expression, but not protein expression, so difficult to say something about the function

- Work flow

  - set up
  - wet lab
  - sequence strategy
  - Bio-info
  - Statistical measures

- What's your purpose? 

  - coding or regulator non-coding
  - reference genome or not?
  - Alternative splicing?
  - which technology can be good for your research question?
  - populations v. specific treatments?

- Stats: biological replication

- Choice of tissue and when you are taking the tissue because, for example, differential gene expression with daylight in plants- circadian rhythm 

- Pool samples in the case of small organisms



- RNA extraction to get good yield:

  - RNAse free environment
  - DNAse treatment
  - get rid of ribosomal RNAs
  - mRNA is enriched with Poly A tail 

- cDNA:

  - RNA is reverse transcribed to DNA (cDNA)

- Library

  - single end
  - paired end

- Choose what kind of sequencing platform:

  - Pyro sequencing by Roche, Ion Torrent
  - GA/hiseq by Illumina
  - Error profiles: incorrect homop…, GC content

- seq coverage: greater than 100 million bp?

- Programming: python, UNIX, R

  ## Amplicon sequencing

  - Targeted approach for analyzing genetic variation in specific genomic regions. 

  - methods

    - library prep- 
      - Extraction 
      - first PCR (to amplify gene of interest)
      - Clean PCR products (running gel or column)
      - send to lab for sequencing or 2nd PCR (barcodes, adaptors), clean, and pool, and sequencing
      - 454 sequencing (obsolete), MiSeq (newer-paired end sequencing) 
    - sequencing
    - data analysis
      - **TRIM**: get your sequence back, and trim off sequence
      - **ALIGN**: if there is a dash it wasn't read but the dash keeps them still aligned, sometimes a dash wasn't added and you need manually add one to keep reads aligned. 
    - Applications

  - | Whole Genome sequencing | RNA seq               | GBS RAD Seq         | Amplicons        |
    | ----------------------- | --------------------- | ------------------- | ---------------- |
    | Everything              | Gene space expressed  |                     | one gene         |
    | single individuals      | few individuals (~93) | Lots of individuals | many individuals |
    |                         |                       |                     |                  |

## GBS (genotyping by sequencing)- (RAD Seq):

- lots of SNPS across the genome, 


- don't care about specific genes, 

- don't need complete genome sequence

- Looking at genetic diversity across multiple individuals

- Getting the SNP genotypes by the process of sequencing (all in one step)

- RAD- restriction assisted DNA sequencing (using restriction enzymes)

- Restriction enzyme- enzyme meant to cut double stranded DNA

  ![](stickyendsDNA.gif)


- BarCodes & Adaptors stick on sticky ends
- When some individuals do not have a restriction enzyme site, and you are only sequencing 100 bp, long fragments will not be sequenced and generate some missing data (trade off). Decide how much missing data you are willing to tolerate. 
- "Reduced Representation" reduces number of reads you have to look at, but might generate missing data. 

------
<div id='id-section5'/>
### Page 5: 2017-02-07

## Getting onto Unix Server...

- Go into the Terminal

- type:  ssh salger@pbio381.uvm.edu

- type 'yes'

- 'top' checks out the serve is being used and by who

- 'q' quits that

- ~/ (~: home directory) 

- 'pwd': present working directory, analogous to "gwd" in R (get working directory)

- 'll': list- to see what is in the directory

- 'mkdir my data': Create a folder 'My Data': 

- 'cd' - change directory, move into that directory 'cs mydata'

- Goal: copy the data from shared file into my directory

  - cd /data/ - go into group shared space, forward slash used to go into original data file, use the ~ to leap back into my personal directory
  - ll- check to see what is in it
  - go into project data. (Do not need forward slash because we are already in the shared directory) (cd project_data)
  - copy into my directory  'cp ssw_sample.txt ~/mydata'
  - move to my directory '~/mydata'
  - use head command to check file 'head ssw_samples.txt'
  - use tail to check end of file 'tail ssw_samples.txt'
  - to check just first 15 lines:
    - head -n 15 saw_samples.txt
  - Grep: Generalized Regular Expressions
  - **grep 'HH' ssw_samples.txt**
  - move to a new folder:
  - **grep 'HH' ssw_samples.txt >ssw_HHonly.txt**
  - move all files ending with only.txt to folder: 
  - **mv *only.txt  sample_by_disease/**

  man grep - to see the help menu for the grep command

- rm (remove file)

- ll -a (shows hidden files)

- To have the command line ask if we want

- typing in a directory, hit tab tab to see what the possibilities are

------
<div id='id-section6'/>
### Page 6: 2017-02-07

## RNA-seq tutorial

Here is the path for raw reads:

```
cd  /data/project_data/fastq/
```

- Samples I'm doing:

  ```
  38_6-18_S_2_R1.fq.gz & 38_6-18_S_2_R2.fq.gz
  ```

Look at the file:

```
zcat 38_6-18_S_2_R1.fq.gz | head
```

There is a script already made that will trim my reads. I will copy this script into my scripts folder...

```
cp  /data/scripts/trim_example.sh .
```

period at the end copies that folder into the folder I am presently in. 

To look at the trim_example script….

```
vim trim_example.sh
```

'i', to insert or edit in Vim

enter the file names and file directory as follows:

```
#!/bin/bash
      java -classpath /data/popgen/Trimmomatic-0.33/trimmomatic-0.33.jar org.usadellab.trimmomatic.TrimmomaticPE \
                -threads 1 \
                -phred33 \
                 /data/project_data/fastq/38_6-18_S_2_R1.fq.gz \
                 /data/project_data/fastq/38_6-18_S_2_R2.fq.gz \
                 /data/project_data/fastq/cleanreads/"38_6-18_S_2_R1_clean_paired.fa" \
                 /data/project_data/fastq/cleanreads/"38_6-18_S_2_R1_clean_unpaired.fa" \
                 /data/project_data/fastq/cleanreads/"38_6-18_S_2_R2_clean_paired.fa" \
                 /data/project_data/fastq/cleanreads/"38_6-18_S_2_R2_clean_unpaired.fa" \
                 ILLUMINACLIP:/data/popgen/Trimmomatic-0.33/adapters/TruSeq3-PE.fa:2:30:10 \
                 LEADING:28 \
             TRAILING:28 \
             SLIDINGWINDOW:6:28 \
             HEADCROP:9 \
             MINLEN:35 \
```

Save the script by clicking shift + colon    wq



Change settings to get executable run, write, read access:

```
chmod u+x trim_example.sh 
```

Running the script:

```
./trim_example.sh
```

Let's take a look in the file.. type  the following and hit tab tab to see what is in it (or go into the cd and then type "ls")



```
cd /data/project_data/fastq/cleanreads/
```



------
<div id='id-section7'/>
### Page 7: 2017-02-08

### Transcriptomics

Why wild systems?

* non model/non traditional model organisms
* silent genes responding to multiple stimuli
* variation between individuals and between populations
* novel trascripts without homologs in closly related model organisms



Microarray: easy for ecological analyses

RNAseq: genome wide ecological tracripts, more involved analyses



1. How much variation is there in gene expression and how is it structured? 288
2. How do environmental stimuli affect expression? 15
3. How does gene expression affect phenotype? 136

______

### Evolutionary Process

* Gene expression heritable <———— Acted on by Natural Selection

* Epigenetics

* Qst -Fst comparisons

* eQTL- expression quantitative trait loci mapping- Id

  * Expression popultions within populations

* Macroevolution

  * drift, selection, bottleneck

    ​
###How does environment affect expression?
* abiotic stress
* Environmental heterogeneity
* host-parasite interactions
* selective biotic and abiotic interactions
* Molecular
* genotype
* phenotypic plasticity




* Flash Freeze

* 'snap shot'

* time course analyses

  * transcriptional responses through time $$$

______

### Alternate phenotypes

* moving from correlation to causation
  * transgenics, RNAi, Crisper/CAS
  * So, if you have a candidate gene associated with a phenotype, you knock it down or out and see if you still get the phenotype

______________

### Future:

RNAseq

Are probes available for the questions we are asking?

Database for proposed ecological annotations?

_____________

### Problems:

bias in signals

Heterogous arrays

polyploidy

RNA pooling

statistical analyses

unannotated genes 

_______-

Something to think about: Expression as a phenotype vs. expression causing a phenotype?

_________

### Paper discussion:

De Panis et al. 2016

Summary of big questions and study system? and what tools used? walk folks through figures

___________________

### Command line stuff!

* skills learning objectives:
* scripts
* paths
  * program
  * imput
  * output
* Filenames
  * in
  * out
* moving through directories/files
* SCP- move from server to your PC
* executing scripts
* calling program

Used to rename a file name:

```
mv filename newfilename
```

Or, can be used to move a file to a new directory

```
mv filename \data...filename
```

Cleaned file names are here….

```
cd /data/project_data/fastq/cleanreads
```

name of clean file names:

```
38_6-18_S_2_R1_clean_paired.fa
38_6-18_S_2_R1_clean_unpaired.fa
38_6-18_S_2_R2_clean_paired.fa
38_6-18_S_2_R2_clean_unpaired.fa
```

Use fastqc to check out quality of the files… (check before and after cleaning)

```
fastqc 38_6-18_S_2_R1_clean_paired.fa
fastqc 38_6-18_S_2_R1.fq.gz
```

**Need to navigate to the correct folder where the file lives to check...

use pwd to check where you are...



Next, open up a new terminal window and DO not sign onto the server, stay on your local drive (desktop, or somewhere better)… Now, move the html file (that was created with fastqc) to local drive with this command:

```
scp salger@pbio381.uvm.edu:/data/project_data/fastq/38_6-18_S_2_R1_fastqc.html ~/Desktop/
```

Now, find your file and check it out!

![](Fastqc1.png)



------
<div id='id-section8'/>
### Page 8: 2017-02-13

## Transcriptomics

enables DE examination (inter- population individual)

* disease resistance
* mating behavior
* adaptive significance

molecular mechanisms -> phenotypic/behavioral plasticity (migration patterns)

Limitations: ref genome quality, gene annotation availability, expense per sample lib. prep



Issues: Under utilization of biological replicates

* requiring independent library preparations
* doesn't included pooled samples
* 23/158 sequencing depth over replication
* derive broad biological conclusions.
* prioritize sequence depth over replication <- problem
* sequence coverage: the average # of reads that align/cover known reference bases
* read depth: Total number of bases sequenced/aligned at given reference base position
* wide dynamic range of RNA-seq data-> noisy (poisson counting error), (technical variance), (biological variance). Biological variance

### General rules of thumb

* use more biological replicates instead of depth
* sequence depth >10 reads/trasncript- 10 reads
* 10-20 million mapped reads/sample
* 3 biological replicates per condition
* conduct a pilot experiment (what is best and most powerful experiment I can afford?)
* What is the smallest fold change I can detect?

______

#### Working with our data:

Useful command to allow command to run when your computer is off... 

```
screen
```

opens a new window.

press control + a + d

```
screen -r
```

___________

# Making a reference transcriptome and Mapping reads to the reference transcriptome

Recall that the general RNAseq data processing work flow is to:

1. Clean and evaluate reads (.fastq)
2. Make and evaluate a transcriptome assembly (.fasta)
3. Map cleaned reads to the transcriptome assembly (makes .sam files)
4. From these sequence alignment files, we can extract two types of information: (a) read counts - the number of reads that uniqely map to each “gene” and (b) single nucleotide polymorphisms between a sample and the reference.
5. With these two types of data, we can go on to differential gene expression analyses and population genomics.

Settling on a high quality reference transcriptome is an iterative process that requires testing different assembly parameters and inputs and evaluating the quality several ways. A quick way for us to move forward with our data analyses for this course, however, is to predict open reading frames (ORFs - include start and stop codons) and keep only transcripts that are at least 100 amino acids long. To do this we can use the program TransDecoder.

Download TransDecoder to use to predict longest Open Reading Frames (ORFs)

```
wget https://github.com/TransDecoder/TransDecoder/archive/v3.0.1.zip
```

```
$ cd /data/project_data/assembly/
$ /data/popgen/TransDecoder-3.0.1/TransDecoder.LongOrfs -t Trinity.fasta 


-first extracting base frequencies, we'll need them later.
CMD: /data/popgen/TransDecoder-3.0.1/util/compute_base_probs.pl Trinity.fasta 0 > Trinity.fasta.transdecoder_dir/base_freqs.dat
CMD: touch Trinity.fasta.transdecoder_dir/base_freqs.dat.ok


- extracting ORFs from transcripts.
-total transcripts to examine: 73435
[73400/73435] = 99.95% done    

#################################
### Done preparing long ORFs.  ###
##################################

    Use file: Trinity.fasta.transdecoder_dir/longest_orfs.pep  for Pfam and/or BlastP searches to enable homology-based coding region identification.

    Then, run TransDecoder.Predict for your final coding region predictions.
```

Evaluate the “longest_orfs.cds” assembly after running Transdecoder.

```
$ /data/popgen/trinityrnaseq-Trinity-v2.3.2/util/TrinityStats.pl longest_orfs.cds


################################
## Counts of transcripts, etc.
################################
Total trinity 'genes':  5693
Total trinity transcripts:  8573
Percent GC: 49.08

########################################
Stats based on ALL transcript contigs:
########################################

    Contig N10: 1626
    Contig N20: 1032
    Contig N30: 780
    Contig N40: 612
    Contig N50: 498

    Median contig length: 390
    Average contig: 518.78
    Total assembled bases: 4447500


#####################################################
## Stats based on ONLY LONGEST ISOFORM per 'GENE':
#####################################################

    Contig N10: 1623
    Contig N20: 1080
    Contig N30: 828
    Contig N40: 642
    Contig N50: 528

    Median contig length: 402
    Average contig: 534.41
    Total assembled bases: 3042405
```

We can also evaluate this assembly by using blastp to compare it to the uniprot_swissprot database.

```
wget https://github.com/Trinotate/Trinotate/releases

# Run the script to download the relevant databases.
/data/popgen/Trinotate-3.0.1/admin/Build_Trinotate_Boilerplate_SQLite_db.pl  Trinotate
```

```
#!/bin/bash/
cd /data/popgen/databases/
makeblastdb -in uniprot_sprot.pep -dbtype prot -out uniprot_sprot
blastp -query /data/project_data/assembly/Trinity.fasta.transdecoder_dir/longest_orfs.pep  -db /data/popgen/databases/uniprot_sprot  -max_target_seqs 1 -outfmt 6 -evalue 1e-5 -num_threads 10 > blastp.outfmt6
```

```
TransDecoder.Predict -t target_transcripts.fasta --retain_blastp_hits blastp.outfmt6
```

These transcriptome assembly processes are ongoing. But for now we will map to the 5,693 “genes” based on the longest ORFs.

Options for improving this assembly include: (1) using more reads from other individuals or trying a different individual, (2) changing the cleaning and assembly parameters. We can evaluate based on the percentage of genes that have good blastp hits and the percentage of single copy orthologs included in the reference (for example using the new program [BUSCO](http://busco.ezlab.org/).

## Map reads from individual samples to reference transcriptome

1. Navigate to the `/data/scripts/` directory to find a script called `bwaaln.sh` that you can
2. copy `cp` to your home directory `~/scritps/` and
3. open with `vim` to edit.
4. You need to enter your “left” reads file name (for those cleaned and paired).
5. Step through the script to make sure you understand each command.

The script looks like this:

```
#!/bin/bash 
 
# To run from present directory and save output: ./bwaaln.sh > output.bwaaln.txt 

myLeft='38_6-24_S_5_R1.fq.gz_left_clean_paired.fq'
echo $myLeft

myRight=${myLeft/_R1.fq.gz_left/_R2.fq.gz_right} 
echo $myRight

myShort=`echo $myLeft | cut -c1-11`
echo $myShort

# bwa index /data/project_data/assembly/longest_orfs.cds  # This only needs to be done once on the reference

bwa aln /data/project_data/assembly/longest_orfs.cds /data/project_data/fastq/cleanreads/$myLeft > $myLeft".sai"
bwa aln /data/project_data/assembly/longest_orfs.cds /data/project_data/fastq/cleanreads/$myRight > $myRight".sai"
bwa sampe -r '@RG\tID:'"$myShort"'\tSM:'"$myShort"'\tPL:Illumina' \
        -P /data/project_data/assembly/longest_orfs.cds $myLeft".sai" $myRight".sai" \
        /data/project_data/fastq/cleanreads/$myLeft \
        /data/project_data/fastq/cleanreads/$myRight > $myShort"_bwaaln.sam"
```

This script could also be made into a loop to map reads of all files one after another to the transcriptome. We could also tidy up file names using `mv` or `rename`.

## You’ve made a Sequence AlignMent (SAM) file!

A SAM file is a tab delimited text file that stores information about the alignment of reads in a FASTQ file to a reference genome or transcriptome. For each read in a FASTQ file, there’s a line in the SAM file that includes

- the read, aka. query, name,
- a FLAG (number with information about mapping success and orientation and whether the read is the left or right read),
- the reference sequence name to which the read mapped
- the leftmost position in the reference where the read mapped
- the mapping quality (Phred-scaled)
- a CIGAR string that gives alignment information (how many bases Match (M), where there’s an Insertion (I) or Deletion (D))
- an ‘=’, mate position, inferred insert size (columns 7,8,9),
- the query sequence and Phred-scaled quality from the FASTQ file (columns 10 and 11),
- then Lots of good information in TAGS at the end, if the read mapped, including whether it is a unique read (XT:A:U), the number of best hits (X0:i:1), the number of suboptimal hits (X1:i:0).

The left (R1) and right (R2) reads alternate through the file. SAM files usually have a header section with general information where each line starts with the ‘@’ symbol. SAM and BAM files contain the same information; SAM is human readable and BAM is in binary code and therefore has a smaller file size.

Find the official Sequence AlignMent file documentation can be found [here](http://samtools.github.io/hts-specs/SAMv1.pdf).

[This BWA man page](http://bio-bwa.sourceforge.net/bwa.shtml) also discusses SAM alignment format and BWA specific optional fields.

- [Some FLAGs to know](http://seqanswers.com/forums/showthread.php?t=17314) - for example what do the numbers in the second column of data mean? [Here’s a SAM FLAG decoder](https://broadinstitute.github.io/picard/explain-flags.html) by the Broad Institute.
- What about the map quality score, MapQ? That’s important! [Here’s a reference](http://www.acgt.me/blog/2014/12/16/understanding-mapq-scores-in-sam-files-does-37-42).

------
<div id='id-section9'/>
### Page 9: 2017-02-15

## Transcriptomes 

Extract expression

What's going on in the background?

### Continuing  with Sequence Alignment files and Read count extraction



#### Let’s check out our .sam files! Try `head` and `tail`.

```
tail -n 100 38_6-18_S_2_bwaaln.sam > tail.sam
vim tail.sam

:set nowrap
```



## You’ve made a Sequence AlignMent (SAM) file!

A SAM file is a tab delimited text file that stores information about the alignment of reads in a FASTQ file to a reference genome or transcriptome. For each read in a FASTQ file, there’s a line in the SAM file that includes

- the read, aka. query, name,
- a FLAG (number with information about mapping success and orientation and whether the read is the left or right read),
- the reference sequence name to which the read mapped
- the leftmost position in the reference where the read mapped
- the mapping quality (Phred-scaled)
- a CIGAR string that gives alignment information (how many bases Match (M), where there’s an Insertion (I) or Deletion (D))
- an ‘=’, mate position, inferred insert size (columns 7,8,9),
- the query sequence and Phred-scaled quality from the FASTQ file (columns 10 and 11),
- **then Lots of good information in TAGS at the end, if the read mapped, including whether it is a unique read (XT:A:U), the number of best hits (X0:i:1), the number of suboptimal hits (X1:i:0).**



BWA generates the following optional fields. Tags starting with ‘X’ are specific to BWA.

> | **Tag**                                  | **Meaning**                              |
> | ---------------------------------------- | ---------------------------------------- |
> |                                          |                                          |
> |                                          |                                          |
> | **NM**                                   | Edit distance                            |
> | **MD**                                   | Mismatching positions/bases              |
> | **AS**                                   | Alignment score                          |
> | **BC**                                   | Barcode sequence                         |
> |                                          |                                          |
> |                                          |                                          |
> | **X0**                                   | Number of best hits                      |
> | **X1**                                   | Number of suboptimal hits found by BWA   |
> | **XN**                                   | Number of ambiguous bases in the referenece |
> | **XM**                                   | Number of mismatches in the alignment    |
> | **XO**                                   | Number of gap opens                      |
> | **XG**                                   | Number of gap extentions                 |
> | **XT**                                   | Type: Unique/Repeat/N/Mate-sw            |
> | **XA**                                   | Alternative hits; format: (chr,pos,CIGAR,NM;)* |
> |                                          |                                          |
> |                                          |                                          |
> | **XS**                                   | Suboptimal alignment score               |
> | **XF**                                   | Support from forward/reverse alignment   |
> | Why is it important to consider whether a read maps uniquely (i.e., to one place in the transcriptome) for gene expression studies? | Number of supporting seeds               |



#### Let’s see how many of our reads map uniquely.

Why is it important to consider whether a read maps uniquely (i.e., to one place in the transcriptome) for gene expression studies?

```
$ grep -c XT:A:U 38_6-18_S_2_bwaaln.sam  
1177827

$ grep -c X0:i:1 38_6-18_S_2_bwaaln.sam 
1182952
```



- "sed" find and replace command

- "-i" find and replace right in the command line

- "s" search

- "/::/" search for two colons

- "/ \ _ /" replace with an underscore

- g = option

- ```
  sed -i `s/::/\_/g' 38_6-18_S_2_bwaaln.sam
  ```

  ​

------
<div id='id-section10'/>
### Page 10:
------
<div id='id-section11'/>
### Page 11:
------
<div id='id-section12'/>
### Page 12:
------
<div id='id-section13'/>
### Page 13:
------
<div id='id-section14'/>
### Page 14:
------
<div id='id-section15'/>
### Page 15:
------
<div id='id-section16'/>
### Page 16:
------
<div id='id-section17'/>
### Page 17:
------
<div id='id-section18'/>
### Page 18:
------
<div id='id-section19'/>
### Page 19:
------
<div id='id-section20'/>
### Page 20:
------
<div id='id-section21'/>
### Page 21:
------
<div id='id-section22'/>
### Page 22:
------
<div id='id-section23'/>
### Page 23:
------
<div id='id-section24'/>
### Page 24:
------
<div id='id-section25'/>
### Page 25:
------
<div id='id-section26'/>
### Page 26:
------
<div id='id-section27'/>
### Page 27:
------
<div id='id-section28'/>
### Page 28:
------
<div id='id-section29'/>
### Page 29:
------
<div id='id-section30'/>
### Page 30:
------
<div id='id-section31'/>
### Page 31:
------
<div id='id-section32'/>
### Page 32:
------
<div id='id-section33'/>
### Page 33:
------
<div id='id-section34'/>
### Page 34:
------
<div id='id-section35'/>
### Page 35:
------
<div id='id-section36'/>
### Page 36:
------
<div id='id-section37'/>
### Page 37:
------
<div id='id-section38'/>
### Page 38:
------
<div id='id-section39'/>
### Page 39:
------
<div id='id-section40'/>
### Page 40:
------
<div id='id-section41'/>
### Page 41:
------
<div id='id-section42'/>
### Page 42:
------
<div id='id-section43'/>
### Page 43:
------
<div id='id-section44'/>
### Page 44:
------
<div id='id-section45'/>
### Page 45:
------
<div id='id-section46'/>
### Page 46:
------
<div id='id-section47'/>
### Page 47:
------
<div id='id-section48'/>
### Page 48:
------
<div id='id-section49'/>
### Page 49:
------
<div id='id-section50'/>
### Page 50:
------
<div id='id-section51'/>
### Page 51:
------
<div id='id-section52'/>
### Page 52:
------
<div id='id-section53'/>
### Page 53:
------
<div id='id-section54'/>
### Page 54:
------
<div id='id-section55'/>
### Page 55:
------
<div id='id-section56'/>
### Page 56:
------
<div id='id-section57'/>
### Page 57:
------
<div id='id-section58'/>
### Page 58:
------
<div id='id-section59'/>
### Page 59:
------
<div id='id-section60'/>
### Page 60:
------
<div id='id-section61'/>
### Page 61:
------
<div id='id-section62'/>
### Page 62:
------
<div id='id-section63'/>
### Page 63:
------
<div id='id-section64'/>
### Page 64:
------
<div id='id-section65'/>
### Page 65:
------
<div id='id-section66'/>
### Page 66:
------
<div id='id-section67'/>
### Page 67:
------
<div id='id-section68'/>
### Page 68:
------
<div id='id-section69'/>
### Page 69:
------
<div id='id-section70'/>
### Page 70:
------
<div id='id-section71'/>
### Page 71:
------
<div id='id-section72'/>
### Page 72:
------
<div id='id-section73'/>
### Page 73:
------
<div id='id-section74'/>
### Page 74:
------
<div id='id-section75'/>
### Page 75:
------
<div id='id-section76'/>
### Page 76:
------
<div id='id-section77'/>
### Page 77:
------
<div id='id-section78'/>
### Page 78:
------
<div id='id-section79'/>
### Page 79:
------
<div id='id-section80'/>
### Page 80:
------
<div id='id-section81'/>
### Page 81:
------
<div id='id-section82'/>
### Page 82:
------
<div id='id-section83'/>
### Page 83:
------
<div id='id-section84'/>
### Page 84:
------
<div id='id-section85'/>
### Page 85:
------
<div id='id-section86'/>
### Page 86:
------
<div id='id-section87'/>
### Page 87:
------
<div id='id-section88'/>
### Page 88:
------
<div id='id-section89'/>
### Page 89:
------
<div id='id-section90'/>
### Page 90:
------
<div id='id-section91'/>
### Page 91:
------
<div id='id-section92'/>
### Page 92:
------
<div id='id-section93'/>
### Page 93:
------
<div id='id-section94'/>
### Page 94:
------
<div id='id-section95'/>
### Page 95:
------
<div id='id-section96'/>
### Page 96:
------
<div id='id-section97'/>
### Page 97:
------
<div id='id-section98'/>
### Page 98:
------
<div id='id-section99'/>
### Page 99:
------
<div id='id-section100'/>
### Page 100:
------
<div id='id-section101'/>
### Page 101:
------
<div id='id-section102'/>
### Page 102:
------
<div id='id-section103'/>
### Page 103:
------
<div id='id-section104'/>
### Page 104:
------
<div id='id-section105'/>
### Page 105:
------
<div id='id-section106'/>
### Page 106:
------
<div id='id-section107'/>
### Page 107:
------
<div id='id-section108'/>
### Page 108:
------
<div id='id-section109'/>
### Page 109:
------
<div id='id-section110'/>
### Page 110:
------
<div id='id-section111'/>
### Page 111:
------
<div id='id-section112'/>
### Page 112:
------
<div id='id-section113'/>
### Page 113:
------
<div id='id-section114'/>
### Page 114:
------
<div id='id-section115'/>
### Page 115:
------
<div id='id-section116'/>
### Page 116:
------
<div id='id-section117'/>
### Page 117:
------
<div id='id-section118'/>
### Page 118:
------
<div id='id-section119'/>
### Page 119:
------
<div id='id-section120'/>
### Page 120:
------
<div id='id-section121'/>
### Page 121:
------
<div id='id-section122'/>
### Page 122:
------
<div id='id-section123'/>
### Page 123:
------
<div id='id-section124'/>
### Page 124:
------
<div id='id-section125'/>
### Page 125:
------
<div id='id-section126'/>
### Page 126:
------
<div id='id-section127'/>
### Page 127:
------
<div id='id-section128'/>
### Page 128:
------
<div id='id-section129'/>
### Page 129:
------
<div id='id-section130'/>
### Page 130:
------
<div id='id-section131'/>
### Page 131:
------
<div id='id-section132'/>
### Page 132:
------
<div id='id-section133'/>
### Page 133:
------
<div id='id-section134'/>
### Page 134:
------
<div id='id-section135'/>
### Page 135:
------
<div id='id-section136'/>
### Page 136:
------
<div id='id-section137'/>
### Page 137:
------
<div id='id-section138'/>
### Page 138:
------
<div id='id-section139'/>
### Page 139:
------
<div id='id-section140'/>
### Page 140:
------
<div id='id-section141'/>
### Page 141:
------
<div id='id-section142'/>
### Page 142:
------
<div id='id-section143'/>
### Page 143:
------
<div id='id-section144'/>
### Page 144:
------
<div id='id-section145'/>
### Page 145:
------
<div id='id-section146'/>
### Page 146:
------
<div id='id-section147'/>
### Page 147:
------
<div id='id-section148'/>
### Page 148:
------
<div id='id-section149'/>
### Page 149:
------
<div id='id-section150'/>
### Page 150:
------
<div id='id-section151'/>
### Page 151:
------
<div id='id-section152'/>
### Page 152:
------
<div id='id-section153'/>
### Page 153:
------
<div id='id-section154'/>
### Page 154:
------
<div id='id-section155'/>
### Page 155:
------
<div id='id-section156'/>
### Page 156:
------
<div id='id-section157'/>
### Page 157:
------
<div id='id-section158'/>
### Page 158:
------
<div id='id-section159'/>
### Page 159:
------
<div id='id-section160'/>
### Page 160:
------
<div id='id-section161'/>
### Page 161:
------
<div id='id-section162'/>
### Page 162:
------
<div id='id-section163'/>
### Page 163:
------
<div id='id-section164'/>
### Page 164:
------
<div id='id-section165'/>
### Page 165:
------
<div id='id-section166'/>
### Page 166:
------
<div id='id-section167'/>
### Page 167:
------
<div id='id-section168'/>
### Page 168:
------
<div id='id-section169'/>
### Page 169:
------
<div id='id-section170'/>
### Page 170:
------
<div id='id-section171'/>
### Page 171:
------
<div id='id-section172'/>
### Page 172:
------
<div id='id-section173'/>
### Page 173:
------
<div id='id-section174'/>
### Page 174:
------
<div id='id-section175'/>
### Page 175:
------
<div id='id-section176'/>
### Page 176:
------
<div id='id-section177'/>
### Page 177:
------
<div id='id-section178'/>
### Page 178:
------
<div id='id-section179'/>
### Page 179:
------
<div id='id-section180'/>
### Page 180:
------
<div id='id-section181'/>
### Page 181:
------
<div id='id-section182'/>
### Page 182:
------
<div id='id-section183'/>
### Page 183:
------
<div id='id-section184'/>
### Page 184:
------
<div id='id-section185'/>
### Page 185:
------
<div id='id-section186'/>
### Page 186:
------
<div id='id-section187'/>
### Page 187:
------
<div id='id-section188'/>
### Page 188:
------
<div id='id-section189'/>
### Page 189:
------
<div id='id-section190'/>
### Page 190:
------
<div id='id-section191'/>
### Page 191:
------
<div id='id-section192'/>
### Page 192:
------
<div id='id-section193'/>
### Page 193:
------
<div id='id-section194'/>
### Page 194:
------
<div id='id-section195'/>
### Page 195:
------
<div id='id-section196'/>
### Page 196:
------
<div id='id-section197'/>
### Page 197:
------
<div id='id-section198'/>
### Page 198:
------
<div id='id-section199'/>
### Page 199:
------
<div id='id-section200'/>
### Page 200:

